<launch>

<!-- UPDATE THIS FILE FOR FOLLOWING ALGORITHM --> 

<!-- _______________________ SIMULATION NODES & PARAMETER/CONFIG FILES _______________________ -->

    <rosparam command="load" file="$(find simulator_prototype)/config/models/SeaKit.yaml" /> 
    <rosparam command="load" file="$(find simulator_prototype)/config/SimulationParameters.yaml" />
    <rosparam command="load" file="$(find virtual_anchor)/config/VirtualAnchorParameters.yaml" />

    <remap from="virtual_anchor/odom" to="sensors/odom" />

    <param name="hil/robot_description" textfile="$(find simulator_prototype)/config/models/Odin.urdf"/>

	 <node pkg="rviz" type="rviz" name="rviz" args="-d $(find simulator_prototype)/config/rviz/SEAKIT.rviz" /> 
    <node pkg="gui" name="gui_node" type="gui" respawn="false" output="screen">
    </node>

    <node pkg="simulator_prototype" type="hil_simulator_node" name="simulated_vessel" output="screen"/>

    <node pkg="hilsim_virtual_anchor_bridge" type="transform_actuators_msg" name="transform_actuators_msg" output="screen" />


<!--  ______________________________ FOLLOWING ALGORITHM NODES _______________________________ -->

    <!--<node pkg="virtual_anchor" type="anchor_dropper" name="anchor_dropper" output="screen"/>-->

    <!--<node pkg="virtual_anchor" type="tf_odom_to_local_NED" name="tf_odom_to_local_NED" output="screen"/>-->

    <node pkg="following_algorithm" type="thrust_output" name="thrust_output" output="screen"/>

    <node pkg="following_algorithm" type="following_algorithm" name="following_algorithm" output="screen"/>

    <node pkg="rqt_reconfigure" type="rqt_reconfigure" name="rqt_reconfigure" output="screen"/>
 
    <!--<node pkg="virtual_anchor" type="controller_dyn_reconfigure" name="controller_dyn_reconfig" output="screen" />-->


 <!-- ___________________________________ LOG NODE __________________________________________ -->
 <!-- This node logs all the data necessary to show a complete test run in gui -->

 <!-- 
<node pkg="rosbag" type="record" name="record_virtual_anchor"  
args="-o 

/home/maureen/Seakit2017Summer/log/

sensors/odom
virtual_anchor/position
input/actuators_sim
virtual_anchor/controller_errors
virtual_anchor/deactivate
virtual_anchor/set_anchor_parameter
/clock" 
output="screen" />
-->

</launch>
